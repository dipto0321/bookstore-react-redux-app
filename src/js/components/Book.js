import React from 'react';
import PropTypes from 'prop-types';

const Book = (props) => {
  const { handleRemove, book } = props;
  const { id, title, category } = book;

  return (
    <div className="card">
      <div className="card-body">
        <div className="columns">
          <div className="col">
            <h4 className="genre">{category}</h4>
            <h3 className="title">{title}</h3>
            <div className="action">
              <button type="button" onClick={() => handleRemove(id)}>
                Remove
              </button>
            </div>
          </div>
          <div className="col" />
        </div>
      </div>
    </div>
  );
};

Book.propTypes = {
  book: PropTypes.instanceOf(Object).isRequired,
  handleRemove: PropTypes.func.isRequired,
};

export default Book;
