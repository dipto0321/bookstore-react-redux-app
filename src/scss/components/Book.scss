.card {
  position: relative;
  display: flex;
  flex-direction: column;
  word-wrap: break-word;
  background-color: $white;
  background-clip: border-box;
  border: 0.1rem solid $translucent-black;
  border-radius: 0.25rem;
  margin: $intercard 0;
}

.card-body {
  flex: 1 1 auto;
  padding: $card-padding;
}

.columns {
  display: flex;
  flex-wrap: wrap;
}

.book-info {
  flex: 0 0 75%;
  max-width: 100%;
}

.progress {
  max-width: $xlarge * 2;
  padding: $large;
}