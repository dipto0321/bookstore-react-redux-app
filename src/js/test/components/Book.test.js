import React from 'react';
import { shallow } from 'enzyme';
import Book from '../../components/Book';

describe('Book component', () => {
  let wrapper;
  let book;
  let handleRemove;
  beforeEach(() => {
    book = {
      id: 1,
      title: 'Antiquities of the Jews',
      category: 'History',
    };
    handleRemove = jest.fn();
    wrapper = shallow(<Book book={book} handleRemove={handleRemove} />);
  });

  test('it should render the Book component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it should call handleRemove on button click', () => {
    wrapper.find('button').simulate('click');
    expect(handleRemove).toHaveBeenLastCalledWith(book.id);
  });
});
