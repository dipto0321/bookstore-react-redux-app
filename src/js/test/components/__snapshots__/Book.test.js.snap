// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book component it should render the Book component correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="columns"
    >
      <div
        className="book-info"
      >
        <h4
          className="genre"
        >
          History
        </h4>
        <h3
          className="title"
        >
          Antiquities of the Jews
        </h3>
        <p>
          by:Â 
          Josephus
        </p>
        <div
          className="action"
        >
          <button
            className="btn btn-link"
            onClick={[Function]}
            type="button"
          >
            Remove
          </button>
        </div>
      </div>
      <div
        className="progress"
      >
        <Progress />
      </div>
      <div
        className="chapter-info"
      >
        <CurrentChapter
          currentChapter=""
        />
        <ConnectFunction
          id={1}
        />
      </div>
    </div>
  </div>
</div>
`;
